djautotask/sync_rest.py

```diff
@@ -462,17 +480,13 @@ def __init__(self, *args, **kwargs):
         self.completed_date = (timezone.now() - timezone.timedelta(
                 hours=request_settings.get('keep_completed_hours')
             )).strftime('%Y-%m-%dT%H:%M:%S.%fZ')
-        self.api_conditions = [
-            [
-                {
-                    "op": "or",
-                    "operands": [
-                        [self.completed_date_field, self.completed_date, 'gt'],
-                        ['status', models.Status.COMPLETE_ID, 'noteq']
-                    ]
-                }
-            ]
-        ]
+        self.api_conditions = ApiCondition(
+            op='or',
+            operands=[
+                ApiCondition(self.completed_date_field, self.completed_date,
+                             op='>'),
+                ApiCondition('status', models.Status.COMPLETE_ID, '!=')
```

Comment:
Oh,  there is no benefit to use general operator here. The missing `op=` (i.e. `op='!='`)  makes deletion of tickets except those with status `complete`. I will role back this one as you said. It is prone to make a big mistake.